src/styles/common/StyledButton.ts

```diff
@@ -0,0 +1,169 @@
+import styled, { css } from 'styled-components'
+import { variants } from 'styled-theming'
+
+import ColourSheet from '../colours'
+import StyleSheet from '../styles'
+
+const {
+  white,
+  whiteDark,
+  blue,
+  blueDark,
+  successLight,
+  successDark,
+  warningLight,
+  warningDark,
+  dangerLight,
+  dangerDark,
+  bgLight,
+  bgDark,
+} = ColourSheet
+
+const { buttonBorder } = StyleSheet
+
+// Used in stories
+// Good to keep around altough not required
+export enum ButtonVariations {
+  'default',
+  'primary',
+  'secondary',
+  'danger',
+  'success',
+  'warning',
+  'cancel',
+}
+
+// Create our variated Button Theme
+// 'kind' refers to a prop on button
+// <ButtonBase kind="danger" />
+export const ButtonTheme = variants('mode', 'kind', {
```

Comment:
here you can see `styled-theming` `variants(...)` creating the base styling via styled-components for the different `button` tweakable via the `kind` prop (could be renamed to `type` or whatever)

Reply:
in .tsx this would be sth like

```jsx
<CancelButton kind="cancel" />
```
or
```jsx
<WarningButton kind="warning" />
```
etc